### PR TITLE
Feature/farm 445 ui join chat widget

### PR DESCRIPTION
**Pull request by isaacmartinezimv**
[[FARM-445 - Ui join chat widget](https://amidodevelopment.atlassian.net/browse/FARM-445?atlOrigin=eyJpIjoiNDk0YTNkN2JiYzY3NDk1YWI1OThlNzcyMmQ0MjU1N2UiLCJwIjoiaiJ9)]

#### 📲 What

Little widget to use for share whatsApp groups or links 

#### 🤔 Why
		
We need it for Community page		

#### 🛠 How
		
Using new WAMF flutter code style

#### ✅ Acceptance criteria

- [ ] Design Review for UI with BA completed. 
- [ ] Manually tested and verified on Android device/emulator (min. Lollipop 5.1)
- [ ] Showcase video / automation test
- [ ] Passing all tests
- [ ] Rebased/merged with latest changes from development and re-tested
- [ ] Removed unsed comments. TODOs must have JIRA for future work.

#### Coding Standards Checklist
- [ ] unit tests 80% coverage on testable code (functions, methods, class)
- [ ] (Architecture) redux state, reducers, actions, middleware defined under lib/redux
- [ ] ui elements defined under lib/ui. 
- [ ] Strings ready for localisation (e.g. defined in lib/utils/strings.dart)
- [ ] Assets (images/icons path) defined in lib/utils/assets.dart
- [ ] Colors defined in lib/utils/colors.dart
- [ ] Dimensions (ie margins padding) defined in lib/utils/dimens.dart
- [ ] TextStyles defined in lib/utils/styles.dart

Recommended Style Guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

#### 🕵️‍♂️ How to test

Notes for QA

#### Reviewers

@farmsmart/amido @farmsmart/wamf

### CODE REVIEW

File: lib/ui/community/LinkBox.dart [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  static final Color imageContainerColor = Color(0xff25d366);
+  static final double imageSize = 10;
+}
+
+class LinkBoxViewModel {
+  String titleText;
+  String detailText;
+  IconData icon;
+  Function onTap;
+  String image;
```
Comment (albertmiro) [inline lib/ui/community/LinkBox.dart]: These should be final

Comment (LeeMatthewHiggins) [reply lib/ui/community/LinkBox.dart]: 👍

Comment (davidfranquet) [reply lib/ui/community/LinkBox.dart]: Done

**Review by davidfranquet — DISMISSED** [review]

File: lib/ui/community/LinkBox.dart [hunk 2]
```diff
@@ -0,0 +1,130 @@
+import 'package:flutter/material.dart';
+
+class _Constants {
+  static final Color cardBackgroundColor = Color(0xfff5f8fa);
```
Comment (LeeMatthewHiggins) [inline lib/ui/community/LinkBox.dart]: this should be in the style

File: lib/ui/community/LinkBox.dart [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  });
+}
+
+class LinkBox extends StatelessWidget {
+  final LinkBoxViewModel _viewModel;
+
+  const LinkBox({
+    Key key,
+    LinkBoxViewModel viewModel,
+  })  : this._viewModel = viewModel,
```
Comment (LeeMatthewHiggins) [inline lib/ui/community/LinkBox.dart:92]: no style :(

File: lib/ui/community/LinkBox.dart (hunk 2 above)
Comment (LeeMatthewHiggins) [reply lib/ui/community/LinkBox.dart]: we don't want all of the linkboxtohave what's app´s green color 🤦‍♂

Comment (davidfranquet) [reply lib/ui/community/LinkBox.dart]: I think that it shouldn't have viewmodel and style. It's an atom for me. But yes, I will pass a background color and icon color as a param to make it more dynamic

File: lib/ui/community/LinkBox.dart (hunk 3 above)
Comment (davidfranquet) [reply lib/ui/community/LinkBox.dart:92]: https://github.com/farmsmart/farmsmart-flutter/pull/74#discussion_r309608762

File: lib/ui/community/LinkBox.dart (hunk 2 above)
Comment (albertmiro) [reply lib/ui/community/LinkBox.dart]: I am agree with @DavidFD005 in this case, I see that as an Atom as it does contain only Flutter Widgets. But yes, we should need to have something to send these values to the Widget to style it.

Comment (davidfranquet) [reply lib/ui/community/LinkBox.dart]: finally I've made it like a molecule, because the list of params became to large.

Comment (davidfranquet) [reply lib/ui/community/LinkBox.dart]: Images updated

**Review by davidfranquet — COMMENTED** [review]
@LeeMatthewHiggins @albertmiro check it again. I've made this widget like a molecule because the list of params became so large. 

Images updated

**Review by LeeMatthewHiggins — DISMISSED** [review]
awesome

**Review by albertmiro — DISMISSED** [review]

**Review by albertmiro — DISMISSED** [review]

**Review by albertmiro — APPROVED** [review]

**Review by LeeMatthewHiggins — APPROVED** [review]

**Pull request merged**